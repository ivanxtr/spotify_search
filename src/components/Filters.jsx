import React from 'react'

const Filters = ({ showPlaylist, data, filterData, filters }) => {

  const genreFilters = () => {
    return filters.map((filter, index) => (
      <button 
      onClick={() => filterData(filter)}
      key={index}> 
        {filter} 
      </button>
    ))
  }

  const showMenu = () => {
    return data && data[0] && data[0].followers
  }

  return (
    <>
        <div className={ showPlaylist && showMenu() ? "filter-container"  : "filter-container hide" }>
          <div className="filter-order">
            <p> Order By </p>
          </div>
          { filters ? genreFilters() : false }
        </div>
    </>
  )
}

export default Filters
