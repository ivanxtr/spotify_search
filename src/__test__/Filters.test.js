import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import Filters from '../components/Filters';

Enzyme.configure({ adapter: new Adapter() });

const dataFixture = [{
      "followers" : {
        "href" : null,
        "total" : 18399815
      },
      "name" : "Twenty One Pilots",
      "popularity" : 86
    }, 
    {
      "followers" : {
        "href" : null,
        "total" : 318574
      },
      "name" : "TWENTY88",
      "popularity" : 58
    }]

describe('Filters Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Filters />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('snapshot render', () => {
    const component = renderer.create(<Filters />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should display proper content', () => {
    const wrapper = mount(
      <Filters
        showPlaylist={true}
        data={dataFixture}
        filterData={() => {}} 
        filters={["popularity","followers"]}
      />);
    expect(wrapper.find('.filter-container')).toHaveLength(1)
    expect(wrapper.find('.hide')).toHaveLength(0)
    expect(wrapper.find('.filter-order')).toHaveLength(1)
    expect(wrapper.find('.filter-order p').text()).toBe(' Order By ')
    expect(wrapper.find('button')).toHaveLength(2)
  })

  it('should be hidden when empty data is recieved', () => {
    const wrapper = mount(
      <Filters
        showPlaylist={true}
        data={[]}
        filterData={() => {}} 
        filters={["popularity","followers"]}
      />);
    expect(wrapper.find('.filter-container')).toHaveLength(1)
    expect(wrapper.find('.hide')).toHaveLength(1)
    expect(wrapper.find('.filter-order')).toHaveLength(1)
    expect(wrapper.find('button')).toHaveLength(2)
  })
})

