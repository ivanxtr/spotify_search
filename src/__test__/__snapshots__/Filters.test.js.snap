// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Component snapshot render 1`] = `
<div
  className="filter-container hide"
>
  <div
    className="filter-order"
  >
    <p>
       Order By 
    </p>
  </div>
</div>
`;
